docs/_components/modal.md

```diff
@@ -9,34 +9,36 @@ Modals are a blocking element, they should be used sparingly and only when we ab
 Example:
 
 ```html
-<div class="modal">
+<div class="modal" id="myModal" role="dialog" aria-labelledby="dialog-title" aria-describedby="dialog-description">
     <div class="modal__dialog">
         <div class="modal__content">
             <div class="modal__header">
-                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
-                <h4 class="modal__title">A simple example</h4>
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close dialog">&times;</button>
+                <h4 class="modal__title" id="dialog-title">A simple example</h4>
             </div>
             <div class="modal__body">
+                <p id="dialog-description" class="sr-only">SR description goes here</p>
```

Comment:
I think this needs fleshing out with a practical example rather than a placeholder, give developers an idea about what level of detail we would expect to provide here.